include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -128,7 +133,10 @@ class pairwise_alignment_algorithm_banded :
         bool const upper_diagonal_ends_before_last_cell = (upper_diagonal + sequence2_size) < sequence1_size;
         bool const lower_diagonal_ends_behind_last_cell = (-lower_diagonal + sequence1_size) < sequence2_size;
 
-        if (upper_diagonal_ends_before_last_cell || lower_diagonal_ends_behind_last_cell)
+        bool const invalid_band = (lower_diagonal_ends_behind_last_cell && !this->last_column_is_free) || // band ends in last column but does not use free ends,
+                                  (upper_diagonal_ends_before_last_cell && !this->last_row_is_free); // band ends in last row but does not use free ends.
+
+        if (invalid_band)
             throw invalid_alignment_configuration{"The selected band [" + std::to_string(lower_diagonal) + ":" +
                                                   std::to_string(upper_diagonal) + "] does not cover the last cell."};
```

Comment:
Same here, this conflicts with https://github.com/seqan/seqan3/pull/1935/files#diff-a374649bb6a2078f62e2bbc24c7014ffL126-L134

What is your ultimate version?